platforms/access-bridge/src/errors.ts

```diff
@@ -0,0 +1,205 @@
+
+  get description(): string {
+    return this.descriptionOverride || this.defaultDescription;
+  }
+}
+
+/**
+ * Factory for generating error classes.
+ * @param errorCode - The error code for errors of this type.
+ * @param statusCode - The corresponding HTTP status code.
+ * @param defaultDescriptionFactory - A function to fill in the default description based on the error context.
+ * @param headersFactory - A function to fill in the error response headers based on the error context.
+ * @returns a new error class.
+ */
+function createError<T>(
+  errorCode: ErrorCode,
+  statusCode: ErrorStatusCode,
+  defaultDescriptionFactory: (context: T) => string,
+  headersFactory: (context: T) => Iterable<[string, string]> = () => []
+): new (context: T & { description?: string }) => AccessBridgeError {
+  return class extends AccessBridgeError {
+    private readonly context: T;
+
+    constructor(context: T & { description?: string }) {
+      super(context?.description);
+      this.context = context;
+    }
+
+    get code(): ErrorCode {
+      return errorCode;
+    }
+
+    get statusCode(): ErrorStatusCode {
+      return statusCode;
+    }
+
+    get headers(): Iterable<[string, string]> {
+      return headersFactory(this.context);
+    }
+
+    protected get defaultDescription(): string {
+      return defaultDescriptionFactory(this.context);
+    }
+  };
+}
+
+/**
+ * Retrieves the description for a given error code.
+ *
+ * This function replaces placeholders in the error description with the provided `value` and `reason`.
+ *
+ * @param {ErrorCode} code - The error code for which the description is needed.
+ * @param {string} [value] - An optional value to replace the `{value}` placeholder in the description.
+ * @param {string} [reason] - An optional reason to replace the `{reason}` placeholder in the description.
+ * @returns {string} The description corresponding to the error code, with the `{value}` and `{reason}`
+ * placeholders replaced by the provided arguments if applicable.
+ */
+export function getErrorCodeDescription(code: ErrorCode, value?: string, reason?: string): string {
+  let description = ErrorCodeDescription[code];
+
+  if (description.includes('{value}')) {
+    description = description.replace('{value}', value || '');
+  }
+
+  if (description.includes('{reason}')) {
+    description = description.replace('{reason}', reason || '');
+  }
+
+  return description;
+}
+
+/**
+ * Send one or more errors.
+ *
+ * @param res - The response object to send the error to. This ends the response.
+ * @param error - The error to send.
+ * @param errors - Any other errors to send. Their status code must match error's.
+ */
+export function sendErrors(res: ServerResponse, error: AccessBridgeError, ...errors: AccessBridgeError[]) {
+  // All errors must share the same error code
+  const statusCode = error.statusCode;
+  if (errors.find((e) => e.statusCode !== statusCode)) {
```

Comment:
It looks like you're not using the `...errors` param in any of the calls of this method, so I would suggest removing it. Otherwise, you should probably refactor it to prevent this condition from happening. Otherwise you have what is essentially a compile time error causing a runtime error.